base/math.jl

```diff
@@ -103,17 +103,20 @@ function evalpoly(x, p::Tuple)
         _evalpoly(x, p)
     end
 end
+evalpoly(x, ::Tuple{}) = zero(one(x)) # dimensionless zero, i.e. 0 * x^0
 
 evalpoly(x, p::AbstractVector) = _evalpoly(x, p)
 
 function _evalpoly(x, p)
     Base.require_one_based_indexing(p)
     N = length(p)
-    ex = p[end]
+    @inbounds p0 = N == 0 ? reduce_empty_iter(+, p) : p[N]
+    s = oftype(one(x) * p0, p0)
+    N <= 1 && return s
     for i in N-1:-1:1
```

Comment:
```suggestion
    for i in N-1:-1:1
```
If `N <= 1` then `N-1:-1:1` is empty.